src/test/java/com/urswolfer/gerrit/client/rest/http/projects/ProjectsRestClientTest.java

```diff
@@ -208,4 +211,92 @@ public TestListRequest withStart(int start) {
+        private String expectedUrl;
+        private JsonElement mockJsonElement = EasyMock.createMock(JsonElement.class);
+        private GerritRestClient gerritRestClient;
+        private ProjectsParser projectsParser;
+        private BranchInfoParser branchInfoParser;
+        private TagInfoParser tagInfoParser;
+        private String mockJsonString = "{\"name\":\"whatever\"}";
+        private ProjectInput mockProjectInput;
+        private ProjectInfo mockProjectInfo = EasyMock.createMock(ProjectInfo.class);
+
+        public ProjectCreateTestCase(String name) {
+            this.name = name;
+            mockProjectInput = new ProjectInput();
+            mockProjectInput.name = name;
+        }
+
+        public ProjectCreateTestCase expectUrl(String expectedUrl) {
+            this.expectedUrl = expectedUrl;
+            return this;
+        }
+
+        public ProjectCreateTestCase execute() throws Exception {
+            ProjectsRestClient projectsRestClient = getProjectsRestClient();
+            ProjectApi api = projectsRestClient.create(mockProjectInput);
+            return this;
+        }
+
+        public void verify() {
+            EasyMock.verify(gerritRestClient, projectsParser);
+        }
+
+        public ProjectsRestClient getProjectsRestClient() throws Exception {
+            return new ProjectsRestClient(
+                    setupGerritRestClient(),
+                    setupProjectsParser(),
+                    setupBranchInfoParser(),
+                    setupTagInfoParser()
+            );
+        }
+
+        public GerritRestClient setupGerritRestClient() throws Exception {
+            gerritRestClient = EasyMock.createMock(GerritRestClient.class);
+            EasyMock.expect(gerritRestClient.putRequest(expectedUrl,mockJsonString))
+                    .andReturn(mockJsonElement)
+                    .once();
+            EasyMock.replay(gerritRestClient);
+            return gerritRestClient;
+        }
+
+        public ProjectsParser setupProjectsParser() throws Exception {
+            projectsParser = EasyMock.createMock(ProjectsParser.class);
+            EasyMock.expect(projectsParser.generateProjectInput(mockProjectInput))
+                    .andReturn(mockJsonString)
+                    .once();
+            EasyMock.expect(projectsParser.parseSingleProjectInfo(mockJsonElement))
+                    .andReturn(mockProjectInfo)
+                    .once();
+            EasyMock.replay(projectsParser);
+            return projectsParser;
+        }
+
+        public BranchInfoParser setupBranchInfoParser() throws Exception {
+            branchInfoParser = EasyMock.createMock(BranchInfoParser.class);
+            return branchInfoParser;
+        }
+
+        public TagInfoParser setupTagInfoParser() throws Exception {
+            tagInfoParser = EasyMock.createMock(TagInfoParser.class);
+            return tagInfoParser;
+        }
+
+        @Override
+        public String toString() {
```

Comment:
Ignore this comment. It is required for test data providers IIRC (useful output).